app/models/concerns/interval.rb

```diff
@@ -12,7 +12,10 @@ module Interval
     scope :finished, -> { where.not(finished_on: nil) }
     scope :earliest_first, -> { order(started_on: 'asc') }
     scope :latest_first, -> { order(started_on: 'desc') }
-
+    scope :started_before, ->(date) { where(arel_table[:started_on].lt(date)) }
+    scope :started_on_or_after, ->(date) { where(arel_table[:started_on].gteq(date)) }
+    scope :finished_before, ->(date) { where(arel_table[:finished_on].lt(date)) }
+    scope :finished_on_or_after, ->(date) { where(arel_table[:finished_on].gteq(date)) }
```

Comment:
Isn't it more readable using endless ranges?

```suggestion
    scope :started_before, ->(date) { where(started_on: ...date) }
    scope :started_on_or_after, ->(date) { where(started_on: date..) }
    scope :finished_before, ->(date) { where(finished_on: ...date) }
    scope :finished_on_or_after, ->(date) { where(finished_on: date..) }
```

Reply:
Readable yes, but I'm not sure it's more _understandable_. `lt` and `gteq` are easier to understand in my opinion than `..` vs `...` - but the endless ranges are more idiomatic Rails so let's go with them.